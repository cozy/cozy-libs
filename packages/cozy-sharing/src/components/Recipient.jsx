import React, { Component } from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { Spinner, withBreakpoints } from 'cozy-ui/transpiled/react'
import { translate, useI18n } from 'cozy-ui/transpiled/react/I18n'
import { withClient } from 'cozy-client'
import { AvatarPlusX, AvatarLink, Avatar } from './Avatar'

import CompositeRow from 'cozy-ui/transpiled/react/CompositeRow'
import DropdownButton from 'cozy-ui/transpiled/react/DropdownButton'
import ActionMenu, { ActionMenuItem } from 'cozy-ui/transpiled/react/ActionMenu'
import { Text, Caption } from 'cozy-ui/transpiled/react/Text'
import Icon from 'cozy-ui/transpiled/react/Icon'
import { Media, Img, Bd } from 'cozy-ui/transpiled/react/Media'

import styles from './recipient.styl'

import { Contact } from '../models'
import Identity from './Identity'

const MAX_DISPLAYED_RECIPIENTS = 3
const DEFAULT_DISPLAY_NAME = 'Share.contacts.defaultDisplayName'

export const RecipientsAvatars = ({
  recipients,
  link,
  size = 'small-plus',
  className,
  onClick
}) => {
  // we reverse the recipients array because we use `flex-direction: row-reverse` to display them correctly
  // we slice first to clone the original array because reverse() mutates it
  const reversedRecipients = recipients.slice().reverse()
  return (
    <div
      className={classNames(
        styles['recipients-avatars'],
        {
          [styles['--interactive']]: onClick
        },
        className
      )}
      onClick={onClick}
    >
      {link && <AvatarLink size={size} />}
      {recipients.length > MAX_DISPLAYED_RECIPIENTS && (
        <AvatarPlusX
          extraRecipients={reversedRecipients
            .slice(MAX_DISPLAYED_RECIPIENTS)
            .map(recipient => Contact.getDisplayName(recipient))}
          size={size}
        />
      )}
      {reversedRecipients
        .slice(0, MAX_DISPLAYED_RECIPIENTS)
        .map((recipient, idx) => (
          <Avatar
            key={idx}
            text={Contact.getInitials(recipient)}
            size={size}
            textId={Contact.getDisplayName(recipient)}
          />
        ))}
    </div>
  )
}

export const UserAvatar = ({ url, size, ...rest }) => (
  <div className={styles['avatar']}>
    <Avatar
      text={Contact.getInitials(rest)}
      size={size}
      textId={Contact.getDisplayName(rest)}
    />
    <Identity name={Contact.getDisplayName(rest)} details={url} />
  </div>
)

export class Permissions extends Component {
  state = {
    revoking: false,
    isMenuDisplayed: false
  }

  onRevoke = async () => {
    const {
      onRevoke,
      document,
      sharingId,
      index,
      isOwner,
      onRevokeSelf
    } = this.props
    this.setState({ revoking: true })
    if (isOwner) {
      await onRevoke(document, sharingId, index)
    } else {
      await onRevokeSelf(document)
    }

    this.setState({ revoking: false })
  }

  render() {
    const {
      isOwner,
      status,
      instance,
      type,
      documentType,
      client,
      t
    } = this.props
    const { revoking, isMenuDisplayed } = this.state
    const isMe =
      instance !== undefined && instance === client.options.uri && !isOwner
    const shouldShowMenu = !revoking && status !== 'owner' && (isMe || isOwner)

    const showMenu = () => this.setState({ isMenuDisplayed: true })
    const hideMenu = () => this.setState({ isMenuDisplayed: false })

    const buttonRef = React.createRef()
    const permissionIconName = type === 'two-way' ? 'rename' : 'eye'

    return (
      <div>
        {revoking && <Spinner />}
        {!shouldShowMenu && !revoking && (
          <span>{t(`Share.status.${status}`)}</span>
        )}
        {shouldShowMenu && !revoking && (
          <div>
            <DropdownButton onClick={showMenu} ref={buttonRef}>
              {t(`Share.type.${type}`)}
            </DropdownButton>
            {isMenuDisplayed && (
              <ActionMenu
                onClose={hideMenu}
                placement="bottom-end"
                anchorElRef={buttonRef}
                preventOverflow
              >
                <ActionMenuItem
                  left={
                    <Icon
                      icon={permissionIconName}
                      color="var(--primaryTextColor)"
                    />
                  }
                >
                  {t(`Share.type.${type}`)}
                </ActionMenuItem>

                <hr />
                <ActionMenuItem
                  onClick={this.onRevoke}
                  left={<Icon icon="trash" color="var(--pomegranate)" />}
                >
                  <Text className="u-pomegranate">
                    {isOwner
                      ? t(`${documentType}.share.revoke.title`)
                      : t(`${documentType}.share.revokeSelf.title`)}
                  </Text>
                  <Caption>
                    {isOwner
                      ? t(`${documentType}.share.revoke.desc`)
                      : t(`${documentType}.share.revokeSelf.desc`)}
                  </Caption>
                </ActionMenuItem>
              </ActionMenu>
            )}
          </div>
        )}
      </div>
    )
  }
}

const PermissionsWithBreakpoints = withBreakpoints()(
  translate()(withClient(Permissions))
)

const Status = ({ status, isMe, instance }) => {
  const { t } = useI18n()
  const shouldShowStatus = ['pending', 'seen'].includes(status) && !isMe
  const text = shouldShowStatus ? t(`Share.status.${status}`) : instance

  let icon = 'to-the-cloud'
  if (shouldShowStatus && status === 'pending') icon = 'paperplane'
  else if (shouldShowStatus && status === 'seen') icon = 'eye'

  return (
    <Media>
      <Img className={styles['recipient-status-icon']}>
        <Icon icon={icon} size={10} />
      </Img>
      <Bd>
        <Caption>{text}</Caption>
      </Bd>
    </Media>
  )
}

const Recipient = (props, { client, t }) => {
  const { instance, isOwner, status, ...rest } = props
  const isMe =
    (isOwner && status === 'owner') || instance === client.options.uri
  const defaultDisplayName = t(DEFAULT_DISPLAY_NAME)
  const defaultInitials = defaultDisplayName[0].toUpperCase()
  const name = Contact.getDisplayName(rest, defaultDisplayName)

  return (
    <CompositeRow
      dense
      className={classNames(styles['recipient'], 'u-ph-0')}
      primaryText={isMe ? t('Share.recipients.you') : name}
      secondaryText={<Status status={status} isMe={isMe} instance={instance} />}
      image={
        <Avatar
          text={Contact.getInitials(rest, defaultInitials)}
          size="small-plus"
          textId={name}
        />
      }
      right={<PermissionsWithBreakpoints {...props} />}
    />
  )
}

Recipient.contextTypes = {
  client: PropTypes.object.isRequired,
  t: PropTypes.func.isRequired
}

export default Recipient

export const RecipientWithoutStatus = ({ instance, ...rest }) => {
  const name = Contact.getDisplayName(rest)
  return (
    <div className={styles['recipient']}>
      <Avatar
        text={Contact.getInitials(rest)}
        size={'small-plus'}
        textId={name}
      />
      <div className={styles['recipient-ident-status']}>
        <Identity name={name} details={instance} />
      </div>
    </div>
  )
}

export const RecipientPlusX = ({ extraRecipients }, { t }) => (
  <div className={styles['recipient']}>
    <AvatarPlusX
      extraRecipients={extraRecipients.map(recipient =>
        Contact.getDisplayName(recipient)
      )}
    />
    <div className={styles['recipient-ident-status']}>
      <Identity
        name={t('Share.members.otherContacts', extraRecipients.length)}
      />
    </div>
  </div>
)

RecipientPlusX.contextTypes = {
  t: PropTypes.func.isRequired
}
