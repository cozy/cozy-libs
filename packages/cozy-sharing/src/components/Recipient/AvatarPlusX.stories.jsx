import AvatarPlusX from './AvatarPlusX'

const meta = {
  component: AvatarPlusX,
  args: {}
}

export default meta

export const Default = {
  name: 'Default',
  args: {}
}
