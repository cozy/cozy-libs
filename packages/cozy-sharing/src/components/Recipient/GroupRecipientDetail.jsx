import React from 'react'

import { Dialog } from 'cozy-ui/transpiled/react/CozyDialogs'
import Typography from 'cozy-ui/transpiled/react/Typography'
import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'

import { GroupRecipientDetailWithAccess } from './GroupRecipientDetailWithAccess'
import { GroupRecipientDetailWithoutAccess } from './GroupRecipientDetailWithoutAccess'

const GroupRecipientDetail = ({ name, owner, members, onClose }) => {
  const { t } = useI18n()
  const withAccess = members.filter(
    member => !['revoked', 'mail-not-sent'].includes(member.status)
  )
  const withoutAccess = members.filter(member =>
    ['revoked', 'mail-not-sent'].includes(member.status)
  )

  const ownerName = owner.public_name

  return (
    <Dialog
      open
      size="small"
      onClose={onClose}
      title={
        <>
          {name}
          {ownerName ? (
            <Typography variant="caption">
              {t('GroupRecipientDetail.subtitle', { ownerName })}
            </Typography>
          ) : null}
        </>
      }
      content={
        <div className="u-flex u-stack-xs u-flex-column">
          {withAccess.length > 0 ? (
            <GroupRecipientDetailWithAccess withAccess={withAccess} />
          ) : null}
          {withoutAccess.length > 0 ? (
            <GroupRecipientDetailWithoutAccess withoutAccess={withoutAccess} />
          ) : null}
        </div>
      }
    />
  )
}

export { GroupRecipientDetail }
