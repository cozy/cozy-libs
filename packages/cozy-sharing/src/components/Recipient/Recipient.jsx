import PropTypes from 'prop-types'
import React from 'react'

import { useClient } from 'cozy-client'
import Fade from 'cozy-ui/transpiled/react/Fade'
import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import ListItem from 'cozy-ui/transpiled/react/ListItem'
import ListItemIcon from 'cozy-ui/transpiled/react/ListItemIcon'
import ListItemText from 'cozy-ui/transpiled/react/ListItemText'
import Typography from 'cozy-ui/transpiled/react/Typography'

import RecipientAvatar from './RecipientAvatar'
import RecipientConfirm from './RecipientConfirm'
import RecipientPermissions from './RecipientPermissions'
import RecipientStatus from './RecipientStatus'
import { FADE_IN_DURATION } from '../../helpers/recipients'
import { getDisplayName } from '../../models'

const DEFAULT_DISPLAY_NAME = 'Share.contacts.defaultDisplayName'

const Recipient = props => {
  const { t } = useI18n()
  const client = useClient()

  const {
    instance,
    isOwner,
    status,
    recipientConfirmationData,
    verifyRecipient,
    fadeIn,
    ...rest
  } = props

  const isMe =
    (isOwner && status === 'owner') || instance === client.options.uri
  const defaultDisplayName = t(DEFAULT_DISPLAY_NAME)
  const name = getDisplayName(rest, defaultDisplayName)

  const RightPart = recipientConfirmationData ? (
    <RecipientConfirm
      recipientConfirmationData={recipientConfirmationData}
      verifyRecipient={verifyRecipient}
    ></RecipientConfirm>
  ) : (
    <RecipientPermissions {...props} className="u-flex-shrink-0" />
  )

  return (
    <Fade in timeout={fadeIn ? FADE_IN_DURATION : 0}>
      <ListItem disableGutters>
        <ListItemIcon>
          <RecipientAvatar size="small" recipient={props} />
        </ListItemIcon>
        <ListItemText
          primary={
            <Typography className="u-ellipsis" variant="body1">
              {isMe ? t('Share.recipients.you') : name}
            </Typography>
          }
          secondary={
            <RecipientStatus status={status} isMe={isMe} instance={instance} />
          }
        />
        {RightPart}
      </ListItem>
    </Fade>
  )
}

Recipient.propTypes = {
  instance: PropTypes.string,
  isOwner: PropTypes.bool,
  status: PropTypes.string,
  recipientConfirmationData: PropTypes.object,
  verifyRecipient: PropTypes.func
}

export default Recipient
