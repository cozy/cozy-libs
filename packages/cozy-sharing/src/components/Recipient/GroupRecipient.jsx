import React, { useState } from 'react'

import { useClient } from 'cozy-client'
import Fade from 'cozy-ui/transpiled/react/Fade'
import ListItem from 'cozy-ui/transpiled/react/ListItem'
import ListItemIcon from 'cozy-ui/transpiled/react/ListItemIcon'
import ListItemSecondaryAction from 'cozy-ui/transpiled/react/ListItemSecondaryAction'
import ListItemText from 'cozy-ui/transpiled/react/ListItemText'
import Typography from 'cozy-ui/transpiled/react/Typography'
import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'

import { GroupRecipientDetail } from './GroupRecipientDetail'
import { GroupRecipientPermissions } from './GroupRecipientPermissions'
import {
  FADE_IN_DURATION,
  getGroupRecipientSecondaryText
} from '../../helpers/recipients'
import { GroupAvatar } from '../Avatar/GroupAvatar'

const GroupRecipient = props => {
  const { name, members, fadeIn, owner, isOwner } = props
  const { t } = useI18n()
  const client = useClient()
  const { isMobile } = useBreakpoints()

  const [isDetailOpened, setDetailOpened] = useState(false)

  const nbMember = members.length
  const nbMemberReady = members.filter(
    member => !['revoked', 'mail-not-sent'].includes(member.status)
  ).length
  const isUserInsideMembers = members.some(
    member => member.instance === client.options.uri
  )

  const toogleDetailOpened = () => {
    setDetailOpened(!isDetailOpened)
  }

  const closeDetail = () => {
    setDetailOpened(false)
  }

  return (
    <>
      <Fade in timeout={fadeIn ? FADE_IN_DURATION : 0}>
        <ListItem
          button
          onClick={toogleDetailOpened}
          gutters={isMobile ? 'default' : 'double'}
        >
          <ListItemIcon>
            <GroupAvatar size="small" />
          </ListItemIcon>
          <ListItemText
            primary={
              <Typography className="u-ellipsis" variant="body1">
                {name}
              </Typography>
            }
            secondary={getGroupRecipientSecondaryText({
              t,
              nbMember,
              nbMemberReady,
              isUserInsideMembers
            })}
          />
          <ListItemSecondaryAction className={isMobile ? 'u-mr-1' : 'u-mr-2'}>
            <GroupRecipientPermissions
              isUserInsideMembers={isUserInsideMembers}
              {...props}
            />
          </ListItemSecondaryAction>
        </ListItem>
      </Fade>
      {isDetailOpened ? (
        <GroupRecipientDetail
          isOwner={isOwner}
          owner={owner}
          name={name}
          members={members}
          onClose={closeDetail}
        />
      ) : null}
    </>
  )
}

export { GroupRecipient }
