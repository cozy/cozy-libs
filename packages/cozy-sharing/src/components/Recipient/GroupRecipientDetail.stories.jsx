import { GroupRecipientDetail } from './GroupRecipientDetail'

const meta = {
  component: GroupRecipientDetail,
  args: {
    name: 'Family',
    owner: {
      public_name: 'Alice'
    }
  }
}

export default meta

export const Default = {
  name: 'Default',
  args: {
    members: [
      {
        name: 'Alice',
        instance: 'http://alice.cozy.localhost:8080',
        status: 'ready'
      },
      { name: 'marine@google.com', status: 'mail-not-sent' },
      { email: 'Martin', status: 'pending' },
      { name: 'Paul', status: 'revoked' }
    ]
  }
}

export const OnlyWithAccess = {
  name: 'OnlyWithAccess',
  args: {
    members: [
      {
        name: 'Alice',
        instance: 'http://alice.cozy.localhost:8080',
        status: 'ready'
      },
      { name: 'Martin', status: 'pending' }
    ]
  }
}

export const OnlyWithoutAccess = {
  name: 'OnlyWithoutAccess',
  args: {
    members: [
      { name: 'marine@google.com', status: 'mail-not-sent' },
      { name: 'Paul', status: 'revoked' }
    ]
  }
}
