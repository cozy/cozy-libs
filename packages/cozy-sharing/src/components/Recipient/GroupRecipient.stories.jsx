import { GroupRecipient } from './GroupRecipient'

const meta = {
  component: GroupRecipient,
  args: {
    name: 'Family',
    isOwner: true,
    read_only: false,
    members: [
      { status: 'ready' },
      { status: 'mail-not-sent' },
      { status: 'pending' },
      { status: 'revoked' }
    ],
    groupIndex: 0
  }
}

export default meta

export const Default = {
  name: 'Default',
  args: {}
}

export const InstanceInsideMembers = {
  name: 'Instance inside members',
  args: {
    isOwner: false,
    instance: 'http://alice.cozy.localhost:8080',
    members: [
      { status: 'ready', instance: 'http://alice.cozy.localhost:8080' },
      { status: 'mail-not-sent' },
      { status: 'pending' },
      { status: 'revoked' }
    ]
  }
}
