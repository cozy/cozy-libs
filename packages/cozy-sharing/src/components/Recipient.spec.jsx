import React from 'react'
import Recipient from './Recipient'
import AppLike from '../../test/AppLike'
import { createMockClient } from 'cozy-client'
import { render, fireEvent } from '@testing-library/react'

describe('Recipient component', () => {
  const client = createMockClient({})
  client.options = {
    uri: 'foo.mycozy.cloud'
  }

  const setup = props => {
    return render(
      <AppLike client={client}>
        <Recipient {...props} />
      </AppLike>
    )
  }

  let createRangeBackup

  beforeAll(() => {
    createRangeBackup = global.document.createRange
    global.document.createRange = jest.fn(() => ({
      setStart: () => {},
      setEnd: () => {},
      commonAncestorContainer: {
        nodeName: 'BODY',
        ownerDocument: document
      }
    }))
  })

  afterAll(() => {
    global.document.createRange = createRangeBackup
  })

  it('should render if isMe ', () => {
    const { getByText } = setup({
      instance: 'foo.mycozy.cloud',
      status: 'ready',
      type: 'two-way'
    })
    expect(getByText('You')).toBeTruthy()
    expect(getByText('foo.mycozy.cloud')).toBeTruthy()
  })

  it('should match snapshot if isOwner', () => {
    const { getByText } = setup({
      instance: 'foo.mycozy.cloud',
      status: 'ready',
      type: 'two-way',
      isOwner: true
    })
    expect(getByText('You')).toBeTruthy()
    expect(getByText('foo.mycozy.cloud')).toBeTruthy()
  })

  it('should call revokeSelf if I am not the owner but try to revoke myself', async () => {
    const onRevoke = jest.fn()
    const onRevokeSelf = jest.fn()

    const { getByText } = setup({
      instance: 'foo.mycozy.cloud',
      status: 'ready',
      type: 'two-way',
      documentType: 'Files',
      onRevoke,
      onRevokeSelf
    })

    fireEvent.click(getByText('Can Change'))
    fireEvent.click(getByText('Remove me from sharing'))
    expect(onRevoke).not.toBeCalled()
    expect(onRevokeSelf).toBeCalled()
  })

  it('should call revoke if I am the owner of the sharing', async () => {
    const onRevoke = jest.fn()
    const onRevokeSelf = jest.fn()

    const { getByText } = setup({
      instance: 'foo.mycozy.cloud',
      status: 'ready',
      type: 'two-way',
      isOwner: true,
      documentType: 'Files',
      onRevoke,
      onRevokeSelf
    })

    fireEvent.click(getByText('Can Change'))
    fireEvent.click(getByText('Remove from sharing'))
    expect(onRevoke).toBeCalled()
    expect(onRevokeSelf).not.toBeCalled()
  })
})
