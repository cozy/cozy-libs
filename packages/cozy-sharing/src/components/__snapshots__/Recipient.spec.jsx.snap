// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipient component should match snapshot if isMe and type file 1`] = `
<AppLike>
  <TestI18n>
    <BaseI18n
      defaultLang="en"
      dictRequire={[Function]}
      lang="en"
    >
      <Status
        client={
          Object {
            "options": Object {
              "uri": "foo.mycozy.cloud",
            },
          }
        }
        instance="foo.mycozy.cloud"
        status="ready"
        t={[Function]}
        type="file"
      >
        <div
          className="recipient-status u-ml-1"
        >
          <withBreakpoints(MenuAwareMobile)
            buttonClassName="recipient-menu-btn"
            className="recipient-menu"
            itemsStyle={
              Object {
                "maxWidth": "280px",
              }
            }
            popover={true}
            position="right"
            text="Share.type.file"
          >
            <MenuAwareMobile
              breakpoints={
                Object {
                  "isDesktop": true,
                  "isExtraLarge": false,
                  "isLarge": true,
                  "isMedium": false,
                  "isMobile": false,
                  "isSmall": false,
                  "isTablet": false,
                  "isTiny": false,
                }
              }
              buttonClassName="recipient-menu-btn"
              className="recipient-menu"
              itemsStyle={
                Object {
                  "maxWidth": "280px",
                }
              }
              popover={true}
              position="right"
              text="Share.type.file"
            >
              <Menu
                breakpoints={
                  Object {
                    "isDesktop": true,
                    "isExtraLarge": false,
                    "isLarge": true,
                    "isMedium": false,
                    "isMobile": false,
                    "isSmall": false,
                    "isTablet": false,
                    "isTiny": false,
                  }
                }
                buttonClassName="recipient-menu-btn"
                className="recipient-menu"
                itemsStyle={
                  Object {
                    "maxWidth": "280px",
                  }
                }
                popover={true}
                position="right"
                text="Share.type.file"
              >
                <Menu
                  breakpoints={
                    Object {
                      "isDesktop": true,
                      "isExtraLarge": false,
                      "isLarge": true,
                      "isMedium": false,
                      "isMobile": false,
                      "isSmall": false,
                      "isTablet": false,
                      "isTiny": false,
                    }
                  }
                  buttonClassName="recipient-menu-btn"
                  className="recipient-menu"
                  component={null}
                  disabled={false}
                  initialOpen={false}
                  itemsStyle={
                    Object {
                      "maxWidth": "280px",
                    }
                  }
                  label="Share.type.file"
                  onSelect={null}
                  onSelectDisabled={null}
                  popover={true}
                  position="right"
                >
                  <div
                    breakpoints={
                      Object {
                        "isDesktop": true,
                        "isExtraLarge": false,
                        "isLarge": true,
                        "isMedium": false,
                        "isMobile": false,
                        "isSmall": false,
                        "isTablet": false,
                        "isTiny": false,
                      }
                    }
                    className="styles__c-menu___2YTiU styles__c-menu--right___3wxbb recipient-menu"
                  >
                    <Button
                      buttonClassName="recipient-menu-btn"
                      disabled={false}
                      onClick={[Function]}
                      text="Share.type.file"
                    >
                      <button
                        className="c-btn styles__c-menu__btn___18qF8 recipient-menu-btn"
                        disabled={false}
                        onClick={[Function]}
                        role="button"
                      >
                        Share.type.file
                      </button>
                    </Button>
                  </div>
                </Menu>
              </Menu>
            </MenuAwareMobile>
          </withBreakpoints(MenuAwareMobile)>
        </div>
      </Status>
    </BaseI18n>
  </TestI18n>
</AppLike>
`;

exports[`Recipient component should match snapshot if isMe and type folder 1`] = `
<AppLike>
  <TestI18n>
    <BaseI18n
      defaultLang="en"
      dictRequire={[Function]}
      lang="en"
    >
      <Status
        client={
          Object {
            "options": Object {
              "uri": "foo.mycozy.cloud",
            },
          }
        }
        instance="foo.mycozy.cloud"
        status="ready"
        t={[Function]}
        type="folder"
      >
        <div
          className="recipient-status u-ml-1"
        >
          <withBreakpoints(MenuAwareMobile)
            buttonClassName="recipient-menu-btn"
            className="recipient-menu"
            itemsStyle={
              Object {
                "maxWidth": "280px",
              }
            }
            popover={true}
            position="right"
            text="Share.type.folder"
          >
            <MenuAwareMobile
              breakpoints={
                Object {
                  "isDesktop": true,
                  "isExtraLarge": false,
                  "isLarge": true,
                  "isMedium": false,
                  "isMobile": false,
                  "isSmall": false,
                  "isTablet": false,
                  "isTiny": false,
                }
              }
              buttonClassName="recipient-menu-btn"
              className="recipient-menu"
              itemsStyle={
                Object {
                  "maxWidth": "280px",
                }
              }
              popover={true}
              position="right"
              text="Share.type.folder"
            >
              <Menu
                breakpoints={
                  Object {
                    "isDesktop": true,
                    "isExtraLarge": false,
                    "isLarge": true,
                    "isMedium": false,
                    "isMobile": false,
                    "isSmall": false,
                    "isTablet": false,
                    "isTiny": false,
                  }
                }
                buttonClassName="recipient-menu-btn"
                className="recipient-menu"
                itemsStyle={
                  Object {
                    "maxWidth": "280px",
                  }
                }
                popover={true}
                position="right"
                text="Share.type.folder"
              >
                <Menu
                  breakpoints={
                    Object {
                      "isDesktop": true,
                      "isExtraLarge": false,
                      "isLarge": true,
                      "isMedium": false,
                      "isMobile": false,
                      "isSmall": false,
                      "isTablet": false,
                      "isTiny": false,
                    }
                  }
                  buttonClassName="recipient-menu-btn"
                  className="recipient-menu"
                  component={null}
                  disabled={false}
                  initialOpen={false}
                  itemsStyle={
                    Object {
                      "maxWidth": "280px",
                    }
                  }
                  label="Share.type.folder"
                  onSelect={null}
                  onSelectDisabled={null}
                  popover={true}
                  position="right"
                >
                  <div
                    breakpoints={
                      Object {
                        "isDesktop": true,
                        "isExtraLarge": false,
                        "isLarge": true,
                        "isMedium": false,
                        "isMobile": false,
                        "isSmall": false,
                        "isTablet": false,
                        "isTiny": false,
                      }
                    }
                    className="styles__c-menu___2YTiU styles__c-menu--right___3wxbb recipient-menu"
                  >
                    <Button
                      buttonClassName="recipient-menu-btn"
                      disabled={false}
                      onClick={[Function]}
                      text="Share.type.folder"
                    >
                      <button
                        className="c-btn styles__c-menu__btn___18qF8 recipient-menu-btn"
                        disabled={false}
                        onClick={[Function]}
                        role="button"
                      >
                        Share.type.folder
                      </button>
                    </Button>
                  </div>
                </Menu>
              </Menu>
            </MenuAwareMobile>
          </withBreakpoints(MenuAwareMobile)>
        </div>
      </Status>
    </BaseI18n>
  </TestI18n>
</AppLike>
`;

exports[`Recipient component should match snapshot if isOwner 1`] = `
<AppLike>
  <TestI18n>
    <BaseI18n
      defaultLang="en"
      dictRequire={[Function]}
      lang="en"
    >
      <Status
        client={
          Object {
            "options": Object {
              "uri": "foo.mycozy.cloud",
            },
          }
        }
        instance="foo.mycozy.cloud"
        isOwner="true"
        status="ready"
        t={[Function]}
      >
        <div
          className="recipient-status u-ml-1"
        >
          <withBreakpoints(MenuAwareMobile)
            buttonClassName="recipient-menu-btn"
            className="recipient-menu"
            itemsStyle={
              Object {
                "maxWidth": "280px",
              }
            }
            popover={true}
            position="right"
            text="Share.status.ready"
          >
            <MenuAwareMobile
              breakpoints={
                Object {
                  "isDesktop": true,
                  "isExtraLarge": false,
                  "isLarge": true,
                  "isMedium": false,
                  "isMobile": false,
                  "isSmall": false,
                  "isTablet": false,
                  "isTiny": false,
                }
              }
              buttonClassName="recipient-menu-btn"
              className="recipient-menu"
              itemsStyle={
                Object {
                  "maxWidth": "280px",
                }
              }
              popover={true}
              position="right"
              text="Share.status.ready"
            >
              <Menu
                breakpoints={
                  Object {
                    "isDesktop": true,
                    "isExtraLarge": false,
                    "isLarge": true,
                    "isMedium": false,
                    "isMobile": false,
                    "isSmall": false,
                    "isTablet": false,
                    "isTiny": false,
                  }
                }
                buttonClassName="recipient-menu-btn"
                className="recipient-menu"
                itemsStyle={
                  Object {
                    "maxWidth": "280px",
                  }
                }
                popover={true}
                position="right"
                text="Share.status.ready"
              >
                <Menu
                  breakpoints={
                    Object {
                      "isDesktop": true,
                      "isExtraLarge": false,
                      "isLarge": true,
                      "isMedium": false,
                      "isMobile": false,
                      "isSmall": false,
                      "isTablet": false,
                      "isTiny": false,
                    }
                  }
                  buttonClassName="recipient-menu-btn"
                  className="recipient-menu"
                  component={null}
                  disabled={false}
                  initialOpen={false}
                  itemsStyle={
                    Object {
                      "maxWidth": "280px",
                    }
                  }
                  label="Share.status.ready"
                  onSelect={null}
                  onSelectDisabled={null}
                  popover={true}
                  position="right"
                >
                  <div
                    breakpoints={
                      Object {
                        "isDesktop": true,
                        "isExtraLarge": false,
                        "isLarge": true,
                        "isMedium": false,
                        "isMobile": false,
                        "isSmall": false,
                        "isTablet": false,
                        "isTiny": false,
                      }
                    }
                    className="styles__c-menu___2YTiU styles__c-menu--right___3wxbb recipient-menu"
                  >
                    <Button
                      buttonClassName="recipient-menu-btn"
                      disabled={false}
                      onClick={[Function]}
                      text="Share.status.ready"
                    >
                      <button
                        className="c-btn styles__c-menu__btn___18qF8 recipient-menu-btn"
                        disabled={false}
                        onClick={[Function]}
                        role="button"
                      >
                        Share.status.ready
                      </button>
                    </Button>
                  </div>
                </Menu>
              </Menu>
            </MenuAwareMobile>
          </withBreakpoints(MenuAwareMobile)>
        </div>
      </Status>
    </BaseI18n>
  </TestI18n>
</AppLike>
`;

exports[`Recipient component should render if isMe  1`] = `
<AppLike>
  <TestI18n>
    <BaseI18n
      defaultLang="en"
      dictRequire={[Function]}
      lang="en"
    >
      <Status
        client={
          Object {
            "options": Object {
              "uri": "foo.mycozy.cloud",
            },
          }
        }
        instance="foo.mycozy.cloud"
        status="ready"
        t={[Function]}
      >
        <div
          className="recipient-status u-ml-1"
        >
          <withBreakpoints(MenuAwareMobile)
            buttonClassName="recipient-menu-btn"
            className="recipient-menu"
            itemsStyle={
              Object {
                "maxWidth": "280px",
              }
            }
            popover={true}
            position="right"
            text="Share.status.ready"
          >
            <MenuAwareMobile
              breakpoints={
                Object {
                  "isDesktop": true,
                  "isExtraLarge": false,
                  "isLarge": true,
                  "isMedium": false,
                  "isMobile": false,
                  "isSmall": false,
                  "isTablet": false,
                  "isTiny": false,
                }
              }
              buttonClassName="recipient-menu-btn"
              className="recipient-menu"
              itemsStyle={
                Object {
                  "maxWidth": "280px",
                }
              }
              popover={true}
              position="right"
              text="Share.status.ready"
            >
              <Menu
                breakpoints={
                  Object {
                    "isDesktop": true,
                    "isExtraLarge": false,
                    "isLarge": true,
                    "isMedium": false,
                    "isMobile": false,
                    "isSmall": false,
                    "isTablet": false,
                    "isTiny": false,
                  }
                }
                buttonClassName="recipient-menu-btn"
                className="recipient-menu"
                itemsStyle={
                  Object {
                    "maxWidth": "280px",
                  }
                }
                popover={true}
                position="right"
                text="Share.status.ready"
              >
                <Menu
                  breakpoints={
                    Object {
                      "isDesktop": true,
                      "isExtraLarge": false,
                      "isLarge": true,
                      "isMedium": false,
                      "isMobile": false,
                      "isSmall": false,
                      "isTablet": false,
                      "isTiny": false,
                    }
                  }
                  buttonClassName="recipient-menu-btn"
                  className="recipient-menu"
                  component={null}
                  disabled={false}
                  initialOpen={false}
                  itemsStyle={
                    Object {
                      "maxWidth": "280px",
                    }
                  }
                  label="Share.status.ready"
                  onSelect={null}
                  onSelectDisabled={null}
                  popover={true}
                  position="right"
                >
                  <div
                    breakpoints={
                      Object {
                        "isDesktop": true,
                        "isExtraLarge": false,
                        "isLarge": true,
                        "isMedium": false,
                        "isMobile": false,
                        "isSmall": false,
                        "isTablet": false,
                        "isTiny": false,
                      }
                    }
                    className="styles__c-menu___2YTiU styles__c-menu--right___3wxbb recipient-menu"
                  >
                    <Button
                      buttonClassName="recipient-menu-btn"
                      disabled={false}
                      onClick={[Function]}
                      text="Share.status.ready"
                    >
                      <button
                        className="c-btn styles__c-menu__btn___18qF8 recipient-menu-btn"
                        disabled={false}
                        onClick={[Function]}
                        role="button"
                      >
                        Share.status.ready
                      </button>
                    </Button>
                  </div>
                </Menu>
              </Menu>
            </MenuAwareMobile>
          </withBreakpoints(MenuAwareMobile)>
        </div>
      </Status>
    </BaseI18n>
  </TestI18n>
</AppLike>
`;
